Xenos now suffer when the queen dies.

GunHog opened the pull request:
# NERF ALERT!!!!

:warning: :rotating_light: :rotating_light: :rotating_light: :rotating_light: :warning: 

:cl: Gun Hog
tweak: The Alien queen, upon her death, will now severely weaken the remaining alien
forces.
/:cl:

The following changes have been requested by @WJohn:

Upon death of the queen, the remaining aliens will suffer the following effects:
- Disabled Hivemind and Evolution for **4 minutes.**
- A very brief period of slowdown.
- Stuttering.
- Confused movement, making it difficult to escape or attack.
- Jitters, as a confirmation to the crew that the aliens are at their weakest.

It comes with a new screen alert to serve as a reminder! Sprite by @WJohn!
### Intent:

The reasoning behind this pull is to drive home the importance of the queen's role and the aliens' need to defend her - especially the Praetorian. Additionally, this makes killing the queen more of the boss battle it was intended to be. Previously, another queen would simply pop up seconds after the first one died, rendering the monumental task of taking down the original queen entirely moot. 

**Defend your queen. Do not let your queen be a moron and fight alone.**

Screen alert is at the top right.

![screenshot 2016-02-25 21 11 34](https://cloud.githubusercontent.com/assets/4955510/13342397/b882b8f6-dc06-11e5-9815-7604780ce360.png)

Commit:
Queen death vulnerability

- The alien queen's death will now severely weaken the remaining alien
forces.

- Disables Hivemind and Evolving for 4 minutes.

- Confuses and cripples aliens for a short while.

- New screen alert to serve as a reminder! Sprite by WJohn!

Ultimate-Chimera commented:
Noooo muh xenos! ;-; :-1:

WJohn commented:
:+1: 

Thank you so much for the work you put into this.

Ultimate-Chimera commented:
B-but WJohn

VincVsVincent commented:
This actually makes lots of sense, add it

WJohn commented:
Don't worry I've got some buffs in mind for xenos.

After all, you should know by now that the development for xenos is two boo boo bandages for every nasty nerf they get!

ChangelingRain commented on code/modules/mob/living/carbon/alien/humanoid/death.dm:
```diff
@@ -19,3 +19,13 @@
+	if(stat == DEAD)
+		return
+
+	for(var/mob/living/carbon/alien/humanoid/A in living_mob_list)
```
Only humanoid aliens? What about larva? Wouldn't it be possible for a larva to evolve, then be fastracked to queen via plasma transfers from other aliens to bypass this?

GunHog replied:
Oh, crap. I did not put it high enough. Will fix.

Ultimate-Chimera commented:
Yay <3

Ultimate-Chimera commented:
Make xenos evolve faster tbh @WJohn

Core0verload commented:
:+1: Nice change. But you can make it display a "connection severed" message, stun, jitter and confuse everyone with a hive node, not just aliens. Then if someone used alien nodes to spy on alien chat when the queen was killed, they will be hit by the backlash too.

GunHog commented:
Not a bad idea, Core. If WJ likes it, I will do it (it is his PR)!

ChangelingRain commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -167,3 +168,23 @@ proc/alien_type_present(var/alienpath)
+	confused += 30
+	stuttering += 30
+	recent_queen_death = 1
+	spawn(2400) //four minutes
```
Make sure the alien still exists after this long.

GunHog replied:
Righto

WJohn commented:
:+1: Idea, core. I think that'd be neat.
Although you'd have to make them *scream instead of *roar.

GunHog commented:
Xenos do not have a sound for *scream

WJohn commented:
I mean for humans, it'd need a unique exception for them. Humans don't exactly roar.

Ultimate-Chimera commented:
Honestly, remove the disabled evolution because it already takes 4 minutes to evolve.

WJohn commented:
It doesn't prevent larvas from growing up, it prevents drones and praetorians from evolving. That's the whole point, to prevent another queen from happening and to let the crew hunt down the remaining xenos in that time frame.

Zilenan91 commented:
Guess that xeno pref is staying off. Xenos were already painful as hell to play, this only makes it worse.


Also this in no way makes the Queen a boss battle because any amount of time in RnD equals laser cannons which means the queen gets 3 or 4 shot from one singular person with one singular laser cannon.

PKPenguin321 commented:
you cucks
"xenos are underpowered and painful to play as wah"
holy shit have you ever been a xeno

WJohn commented:
You have to be awful to lose as xeno.

Like unbelievably bad to actually somehow die.

Stop being bad and you instantly win against anything and everything.

Core0verload commented:
Then I think var/recent_queen_death would be better as a hive node var.

>I mean for humans, it'd need a unique exception for them.

Also exception for "Your queen has been struck down!". I don't think having a hive node makes you loyal to the hive queen.

Zilenan91 commented:
Oh man you have stunbaton hands. You get 4 shotted by any laser and basically one shotted by any semi-competent burn mix because you take doubled burn damage and can't put yourself out in one roll. The facts are that the crew has more bodies than you can deal with, and no matter what you do, the second they hear xenos the entire armory is passed out and your hive is instantly dead. You can do a lot of damage alone, but that tends to piss people off OOCly. All this really does is make it so you don't want to evolve to Queen or Praetorian at all because the lack of mobility does not in any way make up for what you gain in doing it.

GunHog commented:
You need five shots from a laser cannon to crit a queen from full health, six if she is on weeds.

Ultimate-Chimera commented:
Just make larva evolve faster or something.

GunHog commented:
Also I am not sure how to go about doing the check for hive nodes without snowflaking things up.

Zilenan91 commented:
Making larva evolve faster would be nice. If you don't have weeds you literally have to AFK in a corner for 15 minutes to grow up one level.

WJohn commented:
Faster larva evolve time would be nice, I agree. A problem with xenos is their set-up speed is really really slow because they have no weeds to stand on. Maybe we could halve the time it takes to evolve.

But zilenan, there's MANY tips available in game that tell you how to not suck as a xeno. You don't ever git hit by lasers because you're stupid if you actually put yourself in a situation where that can happen. You have stunbatton hands and leap (which is instant win) and can both abuse existing corners or *make your own battlefield thanks to drones and queens* to force melee. After that it's a matter of exposing everything to space, making liberal use of your facehuggers, and picking your targets/sabotaging the station.

Core0verload commented:
>Also I am not sure how to go about doing the check for hive nodes without snowflaking things up.

```
for(/mob/living/carbon/C in living_mob_list)
    var/obj/item/organ/internal/alien/hivenode/node = getorganslot("hivenode")
    if(istype(node)) // just in case someone would ever add a diffirent node to hivenode slot
        node.recent_queen_death = 1
        C.something blah blah
```

GunHog commented:
Making it REALLY hard to be lazy there, Core!

Zilenan91 commented:
Can't get facehuggers if it takes 5 minutes for them to grow up. And even then you're limited by the Queen who may or may not even be alive anymore due to her immobility and how many she layed in the first place because those things cost a fuckton. Also the issue with "not getting hit by lasers" doesn't exist because you can't really do anything but kill/infest as anything but a Drone, meaning you have to slowly jog towards people attempting to stun them, getting slowed by one laser shot and then killed with the rest. If you attempt to stay in your little hidey hole they're just gonna flamethrower you, or send in so many people you have no hope of winning. If you're in anywhere even remotely open or fighting more than one person you die because you can't handle that many people unless you get insane RnG. If you're alone, you can't kill alone either, because the second you start doing anything xeno-related people are going to scream over comms where you are, whereupon you get swarmed and die unless you're luckily near some vents, in which case I hope you love welded vents because all of them are now closed tighter than a gay man's anus.

Drones as a standalone are good and flexible but they can't kill people because they're so slow, do less damage, and despite all of the walls and shit they can make the walls die in 5 spear hits anyways so they basically aren't even there. You also can't use space to your advantage in any way because jetpacks exist and you literally can't deal with them since sentinels can't aim their attack, so space is offlimits, and attempting to crawl into maint rooms to grow aren't a thing either because everyone knows xenos hive in those rooms and since Queens are so immobile, if they go into a small room like electrical maint they've lost before they've even started since they'll just get swarmed out/flamethrowered, and if they make a space breach to escape maint I hope you like jetpacks because people are going to be swarming around you with them hitting you forever until you die.

Basically what I'm saying is humans have way too many options for killing you, and you're too immobile and too close ranged to ever hope to beat them. This pr takes xenos in a really shitty direction by basing them around the Queen and praetorian who die in 5 seconds the large majority of the time, rather than the supremely more effective and more fun lower castes. I've had some ideas for fixing this but I know that I'm not going to be listened to so I'm just not going to say them.

Zilenan91 commented:
Also even from just the concept who cares if the xenos are crippled for 4 minutes they're just gonna sit in a vent the entire time and then immediately evolve

WJohn commented:
That's still a huge point for the crew. The xenos are in full retreat, you can burn away their nest with very little opposition.

Zilenan91 commented:
Who cares about their nest they'll just wait 5 minutes and grow it back again. The eggs are a moot point too because a drone is just gonna go off z-level and 10 minutes later they'll have more.

Actually, can I talk about this? How queens going off z-level to make their hives is more efficient than staying on the station and having conflict with people? Because it's terrible. Laying eggs is such a painfully slow process that if you're in the 10 minute gap from where you plant them and they grow up, you're down shit's creek without a paddle and will die even faster than if you actually had them because the people attacking you by this point all have biohoods which make them 100% immune to said huggers.

AlexCrimson commented:
If you are playing aliens on /tg/ then you are doing it wrong. Go play on Colonial Marines instead.

Making the entire hive suffer for the shitty play of one bad queen? Terrible mechanic.

WJohn commented:
I think it'd be nice if we do a bit of rebalancing to make the timers much smaller, but that's for a future project. The main point of this PR anyway is to open an attack window for the crew, because xenos can form a pretty impenetrable and slowly advancing death wall otherwise.

Ultimate-Chimera commented:
Wait, we're going to see how this goes in play? <3 Tell me when! Bagil please! <3

PKPenguin321 commented:
Too much autism in this PR discussion to be completely honest

Fox-McCloud commented:
Love it.

Can't give this enough :+1:

Hatfish commented:
Since we're all complaining about xenos in this PR anyways, could someone readd shields blocking the hunter's leap?

RandomMarine commented on code/modules/mob/living/carbon/alien/humanoid/humanoid.dm:
```diff
@@ -167,3 +168,23 @@ proc/alien_type_present(var/alienpath)
+/mob/living/carbon/alien/humanoid/proc/queen_death()
+	if(stat == DEAD)
+		return
+	src << "<span class='userdanger'>Your queen has been struck down!</span>"
```
"Your queen has perished!" May be a better, more universal term. 'Struck down' doesn't quite fit if, for example, an accidental plasma fire killed the queen.

KazeEspada replied:
I can assure you that plasma fire was no accident.

Robustin commented:
If queens weren't basically a free kill I'd like it more.

The nest defenders (drones, sentinels, prae, queen) all kinda suck when the laser spam squad shows up. The insanely robust hunters meanwhile are all designed to be on the offense, and if the queen is blitzed she doesn't even have time to call for help (too busy dodging lasers).

Xenos have a real issue with this. Hunters are OP with infinite no-delay leapstun spam, everything else is target practice as long as the humans attack as a group (they can shake people up from spit or at least suppress any sentinels who move in for a tackle after a successful spit).

WJohn commented:
Robustin have you played xenos in the last 6 months? Those points you raised have changed completely. You also should never have to be actively dodging lasers, you sit around corners with your tail sweep, stun hands, and neurotoxin to instawin any fight. Sentinels have been buffed such that they have more health than hunters, are at human movement speed, and are able to go invisible. Hunters also cannot pounce on crowds of people any more, they have a fairly lengthy delay between them.

Anonmare commented:
There's only one buff I want for xenos

>DECREASE THE TIME IT TAKES FOR SOMEONE TO POP HOLY SHIT
At my last count, it takes someone at least 8 minutes to pop from being hugged (a little less if you kill them after the embryo takes) but the time it takes someone to burst and the slow evolution for larva without weeds is pretty ridicolous.

WJohn commented:
Decreasing all these timers can be done in the future, that's not what this PR is about.

GunHog commented:
WJ do not forget to leave me notes on what I should change beyond the code fixes Joan pointed out. I will get around to it when I can!

Commit:
Queen Dying is now based in the node

- Queen's death is now called on the node itself
- Slightly different effects for humans than aliens

TalkingCactus commented:
Slightly related: You don't have to like where it comes from but feel free to pull anything from https://github.com/TalkingCactus/Citadel-Station-13-6th-Port which includes 90% xeno changes.

Robustin commented:
Just got exposed to xenos again, what a joke. 

You can add this nerf as long as my buff PR's don't get caught in a slog of idiotic comments.

>Tackle is now RNG based

Buh gawd.

GunHog commented:
@Robustin I never changed the tackle ability, if you see it somewhere in my changes, point it out, as it is not intended. 

@TalkingCactus I took a look at that huge PR, and I would encourage @WJohn to have a read. All Xeno related PRs I code are on his behalf, and he has final say on what I upload (barring required code changes that do not deviate from WJ's intent).

What I find most interesting in that PR are the breathing and stomping sounds. It is something that I would love to add, but there may be concerns about stealth.

TalkingCactus commented:
@GunHog I essientially switched from silent super killer death machines to loud and scary sort of robust death machines. All smaller castes gained the ability to use stealth also, which suppresses most sounds.

GunHog commented:
It will be something for WJ to consider. I will probably end up coding it depending on how he feels about their balance, currently.

TalkingCactus commented:
Yeah I wouldn't suggest directly porting any of it over, really. It's not optimized in the slightest and frankly I'm a noob coder.

RemieRichards commented:
@TalkingCactus is the "Aliens make a noise when breathing" thing up for grabs because if done right that could be amazing.

TalkingCactus commented:
@RemieRichards It's all free to use. Dig in.

RemieRichards commented on code/modules/mob/living/carbon/alien/organs.dm:
```diff
@@ -118,6 +119,34 @@
+	spawn(2400) //four minutes
+		if(qdeleted(src) || !owner) //In case the node is deleted or removed from the host
+			return
+		recent_queen_death = 0
```
shouldn't this be above the check for owner?

GunHog replied:
Oh, that makes sense. I will toss that in with the (hopefully) final commit.

Commit:
Moves a check

Moves the owner check to after the recent_queen_death var reset.
Otherwise it probably would not be useable if implanted.

RemieRichards commented:
been 6 days, Gun is OK for merge.
